### PR TITLE
Fix HttpError message formatting and handle None content in validate_response

### PR DESCRIPTION
**Pull request by zhixiangli**
Fix for https://github.com/fsspec/gcsfs/issues/833

## Summary
- Handle `content=None` in `validate_response` to avoid `TypeError` when attempting to parse JSON.
- Handle non-JSON error responses in `validate_response` by using the raw content as the error message instead of failing to parse.
- Add test coverage for `validate_response` with `content=None` and invalid JSON content.

## Test Plan
- [x] Added `test_validate_response_content_none` and `test_validate_response_invalid_json` in `gcsfs/tests/test_retry.py`.
- [x] Verified that these tests pass and correctly raise `HttpError` instead of `TypeError`.

### CODE REVIEW

File: gcsfs/retry.py [hunk 1]
```diff
@@ -114,17 +114,19 @@ def validate_response(status, content, path, args=None):
...
+            if hasattr(content, "decode"):
+                content = content.decode()
+            try:
+                error = json.loads(content)["error"]
+                # Sometimes the error message is a string.
+                if isinstance(error, str):
+                    msg = error
+                else:
+                    msg = error["message"]
+            except json.decoder.JSONDecodeError:
```
Comment (Yonghui-Lee) [inline gcsfs/retry.py:128]: Will it be more robust if we also catch KeyError, TypeError?

For example, If the returned JSON is valid but does not contain the "error" key, a KeyError is raised.

Comment (zhixiangli) [reply gcsfs/retry.py:128]: That is the original implementation and falls outside the scope of this PR. This change is intended to avoid passing None/"" to the JSON parser, which reduces overhead and resolves the issue.

Comment (Yonghui-Lee) [reply gcsfs/retry.py:128]: It’s kind of related. I think getting message from content is not critical to fail, and we can set the message to content or empty.

Edge cases keep coming up. Last year, we had to handle the case where error is a string. Now, we need to handle that content is None. We can catch more errors to fix it for all. For example, passing none can be handled because it raises TypeError.

Comment (zhixiangli) [reply gcsfs/retry.py:128]: The root cause of the TypeError is actually `json.loads(content)` failing when content is None, rather than the `["error"]` key access itself.

Since passing `content=None` is a deliberate design choice in our codebase (used to validate headers before downloading the payload), it represents an expected, valid state rather than an exceptional one. Handling this expected state explicitly with if content: is much cleaner and more efficient than letting the JSON parser intentionally fail and then catching the resulting `TypeError`.

**Review by Yonghui-Lee — APPROVED** [review]
